src/IK/include/BipedalLocomotion/IK/GravityTask.h

```diff
@@ -0,0 +1,214 @@
+     * |:----------------------------------:|:----------------------------:|:----------------------------------------------------------------------------------------------------------------------------------------:|:---------:|
+     * |   `robot_velocity_variable_name`   |            `string`          |                            Name of the variable contained in `VariablesHandler` describing the robot velocity                            |    Yes    |
+     * |                `kp`                |            `double`          |                                               Gain of the distance controller                                                            |    Yes    |
+     * |         `target_frame_name`        |            `string`          |                                       Name of the frame to which apply the gravity task                                                  |    Yes    |
+     * @return True in case of success, false otherwise.
+     */
+    // clang-format on
+    bool
+    initialize(std::weak_ptr<const ParametersHandler::IParametersHandler> paramHandler) override;
+
+    /**
+     * Set the kinDynComputations object.
+     * @param kinDyn pointer to a kinDynComputations object.
+     * @return True in case of success, false otherwise.
+     */
+    bool setKinDyn(std::shared_ptr<iDynTree::KinDynComputations> kinDyn) override;
+
+    /**
+     * Set the set of variables required by the task. The variables are stored in the
+     * System::VariablesHandler.
+     * @param variablesHandler reference to a variables handler.
+     * @note The handler must contain a variable named as the parameter
+     * `robot_velocity_variable_name` stored in the parameter handler. The variable represents the
+     * generalized velocity of the robot. Where the generalized robot velocity is a vector
+     * containing the base spatial-velocity (expressed in mixed representation) and the joints
+     * velocity.
+     * @return True in case of success, false otherwise.
+     */
+    bool setVariablesHandler(const System::VariablesHandler& variablesHandler) override;
+
+    /**
+     * Update the content of the task.
+     * @return True in case of success, false otherwise.
+     */
+    bool update() override;
+
+    /**
+     * @brief Set the desired gravity direction expressed the target frame.
+     * @param desiredGravityDirection The desired gravity direction.
+     *
+     * The input is normalized, unless the norm is too small.
+     */
+    bool setDesiredGravityDirectionInTargetFrame(
+        const Eigen::Ref<const Eigen::Vector3d> desiredGravityDirection);
+
+    /**
+     * @brief Set the feedforward angular velocity expressed in the target frame.
+     * @param feedforwardVelocity The desired feedforward velocity in rad/s.
+     *
+     * Only the first two components are used.
+     */
+    bool
+    setFeedForwardVelocityInTargetFrame(const Eigen::Ref<const Eigen::Vector3d> feedforwardVelocity);
```

Comment:
Can you make it as one unique method named `setSetPoint()` as the other tasks?

Reply:
I have added the method in https://github.com/ami-iit/bipedal-locomotion-framework/pull/717/commits/65c53b2c620cf13167700e8c7d9575c96c646304

I preferred to keep also the other two methods since the name is a bit more informative